src/search/view/SearchViewModel.ts

```diff
@@ -465,11 +465,10 @@ export class SearchViewModel {
 
 	private async loadSearchResults<T extends SearchableTypes>(
 		currentResult: SearchResult,
-		getMoreFromSearch: boolean,
 		startId: Id,
 		count: number,
 	): Promise<{ items: T[]; newSearchResult: SearchResult }> {
-		const result = getMoreFromSearch && hasMoreResults(currentResult) ? await this.searchFacade.getMoreSearchResults(currentResult, count) : currentResult
+		const result = hasMoreResults(currentResult) ? await this.searchFacade.getMoreSearchResults(currentResult, count) : currentResult
```

Comment:
so in this place, we're asking the result (in effect, the index) if there's more results and only then call getMoreSearchResults. getMoreSearchResults has its own check for moreResults.length === 0 and may extend the index if it's true. what's the intended behavior here, is the check to prevent us from accidentally extending the index?

Reply:
I guess that was the original intent? the searchFacade checks moreResults too, and only extends if needed so we may not even need this check here anymore at all, just go ahead and call getMoreSearchResults (which is what is essentially happening right now anyway)